src/api/app/models/user.rb

```diff
@@ -216,29 +216,31 @@ def self.find_with_credentials_via_ldap(login, password)
     user
   end
 
-  # The currently logged in user (might be nil). It's reset after
-  # every request and normally set during authentification
-  def self.current
-    Thread.current[:user]
-  end
-
-  def self.nobody
-    Thread.current[:nobody] ||= find_nobody!
-  end
-
   # Currently logged in user or nobody user if there is no user logged in.
   # Use this to check permissions, but don't treat it as logged in user. Check
   # is_nobody? on the returned object
-  def self.current_or_nobody
+  def self.possibly_nobody
     current || nobody
   end
 
-  def self.current=(user)
-    Thread.current[:user] = user
+  # Currently logged in user. Will thrown an exception if no user is logged in.
+  # So the controller needs to require login if using this (or models using it)
+  def self.required
+    raise ArgumentError, 'Requiring user, but found nobody' unless current && !current.is_nobody?
+    current
+  end
+
+  def self.admin_session?
+    current && current.is_admin?
   end
 
-  def self.current_login
-    current.try(:login) || NOBODY_LOGIN
+  # Currently logged in user or nil
+  def self.not_required
+    current if current && !current.is_nobody?
+  end
+
+  def self.set_current(user)
```

Comment:
Not that I have a strong opinion about this, but in the ruby world the setter would usually be `current=`. We usually try to avoid `set_foo`, `do_bar`

Reply:
I know, basically once it's agreed to go this route I can revert this. But as long as this is in a parallel branch, I need a way to grep for User.current in rebased code :(

Reply:
If you have a suggestion, we can also fixing this by removing current completely and replace it with User.tobedefined=

Reply:
Okay, I see. I don't have a better name right now and if this changes anyway it can stay as is.
Though I would add a fixme so everyone knows and you don't have to answer this again and again.

Reply:
How about User.session= ?

Reply:
@coolo so as I understood, you will do then later a follow-up PR removing this ```set_current``` and moving it to ```self.current=``` right?

Reply:
Unless I get upvotes for User.session= yes, that's the plan.